docker/Dockerfile

```
@@ -1,40 +1,70 @@
+# set path
+ARG PRODUCTION_PATH="/usr/local/app"
+
+# Use official node image as the base image
+FROM node:16-bullseye-slim AS builder
+
+# needed to use PRODUCTION_PATH before the first FROM
+ARG PRODUCTION_PATH
+
+# add the source code to app
+COPY ./ $PRODUCTION_PATH
+
+# set workdir
+WORKDIR $PRODUCTION_PATH
+
+# install for production and cleanup
+RUN npm run deployment
+
+####  building done  ####
+#########################
+####  prod container ####
 # Use official node image as the base image, important to be on debian based for apt-get
 FROM node:16-bullseye-slim
 
+# needed to use PRODUCTION_PATH before the first FROM
+ARG PRODUCTION_PATH
+
 # Disable interactive frontend to prevent install dialogues
 ARG DEBIAN_FRONTEND=noninteractive
+# set production patch
 
-# install nginx
-RUN apt-get update && apt-get upgrade -y && apt-get install nginx -y
+WORKDIR $PRODUCTION_PATH
 
-# add the source code to app
-COPY ./ /usr/local/app/
-
-# set workdir
-WORKDIR /usr/local/app
+# install necessary packages
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install --no-install-recommends \
+    nginx \
+    ca-certificates -y && \
```

```suggestion
        nginx \
        ca-certificates -y && \
```

Does this work? It'd be nicer to read imo.